### ISSUE TITLE
background color detected by color contrast is incorrect in table

### ISSUE THREAD

**Issue by devaradhan** [dequelabs/axe-core#273]
For the html https://codepen.io/demuruge/pen/ObGNqP
axe identifies that the background color for "Cell content" is #007acc where as the acutual background color is #f3f3f3.
In this case, the background color style applied on the tr element is ignored.
document.elementsFromPoint (in https://github.com/dequelabs/axe-core/blob/b45ae65dd2a8a9d7413a622b8f0f879451dac9e7/lib/commons/color/get-background-color.js) doesnt seem to be ignoring tr element.

**Comment by dylanb** [comment]
Here is the relevant HTML

```
<html>
<body>
    <div style="background-color:#007acc">
        <table style="width:100%;">
            <tr style="background-color:#f3f3f3;height:40px;">
            <td style="display: table-cell; color:#007acc">
                Cell content
            </td>
            </tr>
        </table>
    </div>
</body>
</html>
```

**Comment by chriskrycho** [comment]
Seeing similar in an app I'm developing. I have a `button` with actual computed styles (pulled straight from the Firefox inspector) of:

```css
.foo-button {
  color: rgb(255, 255, 255);
  background-color: rgb(207, 31, 40);
}
```

Analysis reports:

> Element has insufficient color contrast of 1.00 (foreground color: #ffffff, background color: #ffffff, font size: 10.5pt, font weight: normal)

The only thing in the whole DOM tree leading to this button which explicitly sets a different background color is the `body` element at the root.

Effective tree:

```html
<html>
<body style='background-color: #ffffff'>
  <main class='main' role='main'>
    <section>
      <form role="search">
        <button style='background-color: #cf1f28; color: #ffffff' type='submit'>Find Locations</button>
      </form>
    </section>
  </main>
</body>
</html>
```

**Comment by vlad-saling** [comment]
I think I have similar issue with aXe Google Chrome plugin. I have a very simple UI, where I set background color and text color on anchor link. Visual representation is as expected: dark teal `#00617a` rectangle with `#fff` text. When I use Chrome's experimental accessibility dev tools it says it has sufficient (AAA level) contrast of 7.02. aXe however flags that element as violation saying: 
```
Ensures the contrast between foreground and background colors meets WCAG 2 AA contrast ratio thresholds.

Element has insufficient color contrast of 1.00 (foreground color: #ffffff, background color: #ffffff, font size: 9.0pt, font weight: normal)
```

What is perhaps most interesting is that aXe doesn't seem to flag that item in all viewport sizes. I have no media queries in style sheet yet, so there is no viewport-size based color/font overrides. Yet, on large enough screen I have no accessibility issues detected by aXe, but when I squeeze viewport to a more narrow view, aXe finds violations like the one mentioned above.

Here's a test URL: http://codepen.io/vlad-saling/pen/KaBErB (just resize the page until you see mentioned issue in aXe report)

Note: I though it could be related to CSS custom variables, but it isn't. Works the same with "hardcoded" color.

**Comment by 1Copenut** [comment]
I've seen this behavior with the Chrome plugin on OSX El Capitan, but only in certain situations. I have a large background image + dark background color that is showing the insufficient color contrast of 1.00, but only when I pin the Chrome dev tools palette to the right of my browser window. If I pin it to the bottom of the window or let it float as a separate window, the error goes away.

In my case, it's definitely an error--the text is #ffffff on a #3e5466 background, for a computed contrast of 7.88, according to Jonathan Snook's colour contrast calculator: https://snook.ca/technical/colour_contrast/colour.html#fg=FFFFFF,bg=3E5466

**Comment by 1Copenut** [comment]
Following onto my previous comment, I am seeing this error come and go, based on browser window width, and placement of the Chrome Dev Tools palette. All examples assume Mac OSX El Capitan, with latest Chrome.

1. 800px width, dev tools docked on right -- Contrast error appears
2. 1,024px width, dev tools docked on right -- Contrast error appears
3. 1,280px width, dev tools docked on right -- Contrast error does not appear (expected behavior)
4. 1,366px width, dev tools docked on right -- Contrast error does not appear (expected behavior)
5. 1,440px width, dev tools docked on right -- Contrast error does not appear (expected behavior)

**Comment by marcysutton** [comment]
Super strange about El Capitan and browser window size. Thanks for the info everyone, this is all very helpful. We'll look into it.

**Comment by dylanb** [comment]
duplicate of https://github.com/dequelabs/axe-core/issues/277

**Comment by liloda** [comment]
I believe the original issue is related to table background color, not browser width. I work with Devaradhan, and the issue appears in several instances regardless of browser size or docking of dev tools. I'm not sure this is a duplicate of #277

**Comment by marcysutton** [comment]
I'll reopen the issue so we can look into it.

**Comment by dylanb** [comment]
@liloda can you provide an example on jsfiddle or codepen?

**Comment by liloda** [comment]
See the codepen link posted by Deva at the beginning of this thread.

**Comment by dylanb** [comment]
@liloda ah, I see now how the confusion happened - my bad!

**Comment by marcysutton** [comment]
I've got the `tr` issue reproduced in a test, I'll commit the fix soon so it can go into our next release. 

Can the rest of you re-test against axe-core 2.2.0 and file new issues for anything that persists? Thanks.
